modules/aggregation-layers/src/aggregation-layer-v9/cpu-aggregator/cpu-aggregator.ts

```diff
@@ -0,0 +1,189 @@
+import type {Aggregator, AggregationProps} from '../aggregator';
+import {_deepEqual as deepEqual, BinaryAttribute} from '@deck.gl/core';
+import {sortBins, packBinIds} from './sort';
+import {aggregateChannel} from './aggregate';
+import {VertexAccessor, evaluateVertexAccessor} from './vertex-accessor';
+
+/** Settings used to construct a new CPUAggregator */
+export type CPUAggregatorSettings = {
+  /** Size of bin IDs */
+  dimensions: number;
+  /** Accessor to map each data point to a bin ID.
+   * If dimensions=1, bin ID should be a number;
+   * If dimensions>1, bin ID should be an array with [dimensions] elements;
+   * The data point will be skipped if bin ID is null.
+   */
+  getBin: VertexAccessor<number | number[] | null, any>;
+  /** Accessor to map each data point to a weight value, defined per channel */
+  getWeight: VertexAccessor<number>[];
+};
+
+/** Options used to run CPU aggregation, can be changed at any time */
+export type CPUAggregationProps = AggregationProps & {};
+
+export type Bin = {
+  id: number | number[];
```

Comment:
Plural?
```suggestion
  ids: number | number[];
```

Reply:
`number[]` is still a single id, when dimensions>1